MBBSEmu.Tests/ExportedModules/Majorbbs/nctime_Tests.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace MBBSEmu.Tests.ExportedModules.Majorbbs
+{
+    public class nctime_Tests : ExportedModuleTestBase
+    {
+        private const int NCTIME_ORDINAL = 430;
+
+        [Theory]
+        [InlineData(12, 4, 10, "12:04:10", true)]
+        [InlineData(3, 9, 2, "03:09:02", false)]
+        [InlineData(4, 11, 16, "04:11:16", true)]
+        [InlineData(7, 5, 4, "07:05:04", false)]
+        [InlineData(10, 1, 28, "10:01:28", true)]
+        [InlineData(14, 3, 36, "14:03:36", false)]
+        [InlineData(18, 12, 8, "18:12:08", true)]
+        [InlineData(23, 1, 44, "23:01:44", false)]
+        [InlineData(23, 59, 58, "23:59:58", true)]
+        public void nctime_Test(int hour, int minutes, int seconds, string expectedTime, bool allocateNCTIME)
+        {
+            //Reset State
+            Reset();
+
+            //Set Argument Values to be Passed In
+            var packedTime = hour << 11 | minutes << 5 | seconds >> 1;
+
+            if (allocateNCTIME)
```

Comment:
could you also tweak the `nctime` implementation and replace `if (!TryGetVariable("NCTIME")) { AllocateVariable(); `

with a single `GetOrAllocateVariablePointer`?